src/drivers/actuators/vertiq_io/vertiq_client_manager.hpp

```diff
@@ -0,0 +1,275 @@
+	* @brief Creates and adds a new entry wrapper object to our array of entry wrappers
+	*
+	* @param px4_param A parameter stored in PX4. This can be found with the param_find function
+	* @param entry A pointer to a Vertiq client entry
+	*/
+	template <typename iquart_data_type , typename px4_data_type>
+	void AddNewClientEntry(param_t px4_param, ClientEntryAbstract *entry){
+		if(_added_entry_wrappers < MAX_CLIENT_ENTRIES){
+			_entry_wrappers[_added_entry_wrappers] = new EntryWrapper<iquart_data_type, px4_data_type>;
+			_entry_wrappers[_added_entry_wrappers]->ConfigureStruct(px4_param, entry);
+			_added_entry_wrappers++;
+		}else{
+			PX4_INFO("Could not add this entry. Maximum number exceeded");
+		}
+	}
+
+private:
+	/**
+	* @brief Initialize all of the Vertiq Clients that we want to use
+	*
+	* @param object_id The object ID with which to initialize our clients
+	*/
+	void InitVertiqClients(uint8_t object_id);
+
+	/**
+	* @brief Initialize all of the Entry Wrappers
+	*/
+	void InitEntryWrappers();
+
+	uint8_t _object_id_now;
+
+	//We need a serial handler in order to talk over the serial port
+	VertiqSerialInterface *_serial_interface;
+
+	//These exist whenever we're using IQUART
+	EntryWrapper<float, float> _velocity_max_entry;
+	EntryWrapper<float, float> _voltage_max_entry;
+	EntryWrapper<uint8_t, int32_t> _control_mode_entry;
+	EntryWrapper<uint8_t, int32_t> _motor_direction_entry;
+	EntryWrapper<uint8_t, int32_t> _fc_direction_entry;
+
+#ifdef CONFIG_USE_IFCI_CONFIGURATION
+	EntryWrapper<uint8_t, int32_t> _throttle_cvi_entry;
+#endif //CONFIG_USE_IFCI_CONFIGURATION
+
+#ifdef CONFIG_USE_PULSING_CONFIGURATION
+	EntryWrapper<uint8_t, int32_t> _pulsing_voltage_mode_entry;
+	EntryWrapper<uint8_t, int32_t> _x_cvi_entry;
+	EntryWrapper<uint8_t, int32_t> _y_cvi_entry;
+	EntryWrapper<float, float> _pulse_zero_angle_entry;
+	EntryWrapper<float, float> _pulse_velo_cutoff_entry;
+	EntryWrapper<float, float> _pulse_torque_offset_entry;
+	EntryWrapper<float, float> _pulse_volt_limit_entry;
+#endif //CONFIG_USE_PULSING_CONFIGURATION
+
+////////////////////////////////////////////////////////////////////////
+//Vertiq client information
+	//Some constants to help us out
+	static const uint8_t MAXIMUM_CONFIGURATION_CLIENTS = 20; //These are clients whose module ID will change when Target Module ID changes
+	static const uint8_t MAXIMUM_OPERATIONAL_CLIENTS = 20; //These are clients that are used for module control/telemetry. They have a static Module ID
+
+	//Client Arrays
+	ClientAbstract *_configuration_client_array[MAXIMUM_CONFIGURATION_CLIENTS];
+	uint8_t _configuration_clients_in_use = 0;
+
+	ClientAbstract *_operational_client_array[MAXIMUM_OPERATIONAL_CLIENTS];
+	uint8_t _operational_clients_in_use = 0;
+
+	//Clients
+	PropellerMotorControlClient _broadcast_prop_motor_control;
+	ArmingHandlerClient _broadcast_arming_handler;
+	IQUartFlightControllerInterfaceClient *_telem_ifci;
+	EscPropellerInputParserClient *_prop_input_parser_client;
+
```

Comment:
This is something that came off of my comment immediately above this, about "why does the telem IFCI live here and the operational IFCI lives in Vertiq IO", and is about the general conceptual/architectural question of  what exactly the goal of the Client Manager is. I just split it off to its own comment so the more specific and more general conversations wouldn't get crossed.

The exact goal of the Client Manager is a bit muddled to me right now. Is it there to be the place where all clients are created and initialized? Right now that's almost the case here, but we also make an IFCI client in Vertiq IO so that doesn't quite work. And when we get to the servo demo we make the multi-turn clients in Vertiq IO and the arming handler stuff in Client Manager, so still spread out there.

Is the goal that Client Manager creates and owns all configuration clients and operational clients come from outside? Right now I would say that isn't quite the case since the IFCI Telem client is created in the Client Manager, but really functions as an operational client, it seems. We have a separate _ifci_client for configuration stuff I believe. And the broadcast prop motor control and arming handler in here are operational as well, but created in client manager. So right now its a mix.

Client manager also does more general client updating stuff, like the Handle Client Communication, and it does that on clients both from within itself and from outside itself.

I guess what I'm getting at right now is when we say its a "ClientManager", what does that mean? Whose clients does it manage, its own or everyone's? And what exactly is its management role? Right now it does some creating, some managing, some handing out clients to other people. 

I'm not interested in any additional functionality, but what I am wondering is if it would be cleaner and clearer conceptually if the Client Manager's responsibilities were split up into smaller more focused pieces. Maybe the "ClientManager" is just responsible for taking in either Configuration or Operational Clients that other people give it, and providing mechanisms for handling their communication/updating them. And separately their may be some "ConfigurationHandler" or whatever that handles making and owning t he configuration clients and handles more configurationy specific things that live in ClientManger now. Then operational stuff just comes from the Vertiq IO and whatever other operational scripts are running. Or maybe we should just clarify that all configuration clients are meant to be created and owned fully within the client manager, but not operational ones. Or maybe its just every client needs to be made here. None of these are fully realized proposals, just illustrations of the kind of clarity and approaches I'm looking for.

To conclude my novel: Basically, I'm a bit muddled on the division of responsibilities and conceptual role of the ClientManager, and either want to clarify and consolidate its current role or spread it out to more focused pieces. It's a bigger more, abstract question then many of my other comments, so probably best if we sit down and talk about it.